cmake/information_helpers.cmake

```
@@ -1,11 +1,104 @@
+            if (GINKGO_INTERFACE_LIBS_LINK_FLAGS)
+                filter_generator_expressions("${GINKGO_INTERFACE_LIBS_LINK_FLAGS}"
+                    GINKGO_INTERFACE_LIB_NAME)
+                list(APPEND GINKGO_INTERFACE_LIBS_FOUND "${GINKGO_INTERFACE_LIB_NAME}")
+            elseif (NOT "${GINKGO_INTERFACE_LIB_NAME}" IN_LIST GINKGO_INTERFACE_LIBS_FOUND)
+                list(APPEND GINKGO_INTERFACE_LIBS_FOUND "${GINKGO_INTERFACE_LIB_NAME}")
+            endif()
+
+            # Populate the include directories
+            get_target_property(GINKGO_LIBS_INTERFACE_INCS "${_libs}"
+                INTERFACE_INCLUDE_DIRECTORIES)
+            foreach(_incs ${GINKGO_LIBS_INTERFACE_INCS})
+                filter_generator_expressions("${_incs}" GINKGO_INTERFACE_INC_FILTERED)
+                if (GINKGO_INTERFACE_INC_FILTERED AND NOT
+                        "-I${GINKGO_INTERFACE_INC_FILTERED}" IN_LIST GINKGO_INTERFACE_CFLAGS_FOUND)
+                    list(APPEND GINKGO_INTERFACE_CFLAGS_FOUND "-I${GINKGO_INTERFACE_INC_FILTERED}")
+                endif()
+            endforeach()
+
+            # Populate the compiler options and definitions if needed
+            get_target_property(GINKGO_LIBS_INTERFACE_DEFS "${_libs}"
+                INTERFACE_COMPILE_DEFINITIONS)
+            if (GINKGO_LIBS_INTERFACE_DEFS)
+                list(APPEND GINKGO_INTERFACE_CFLAGS_FOUND "${GINKGO_LIBS_INTERFACE_DEFS}")
+            endif()
+            get_target_property(GINKGO_LIBS_INTERFACE_OPTS "${_libs}"
+                INTERFACE_COMPILE_OPTIONS)
+            filter_generator_expressions("${GINKGO_LIBS_INTERFACE_OPTS}" GINKGO_LIBS_INTERFACE_OPTS_FILTERED)
+            if (GINKGO_LIBS_INTERFACE_OPTS)
+                list(APPEND GINKGO_INTERFACE_CFLAGS_FOUND "${GINKGO_LIBS_INTERFACE_OPTS_FILTERED}")
+            endif()
+
+            # Keep recursing through the libraries
+            get_target_property(GINKGO_LIBS_INTERFACE_LIBS "${_libs}"
+                INTERFACE_LINK_LIBRARIES)
+            ginkgo_interface_libraries_recursively("${GINKGO_LIBS_INTERFACE_LIBS}")
+        elseif(EXISTS "${_libs}")
+            list(APPEND GINKGO_INTERFACE_LIBS_FOUND "${_libs}")
+        endif()
+    endforeach()
+endmacro()
+
 macro(ginkgo_interface_information)
+    set(GINKGO_INTERFACE_LINK_FLAGS "-L${CMAKE_INSTALL_PREFIX}/${GINKGO_INSTALL_LIBRARY_DIR}")
+    unset(GINKGO_INTERFACE_LIBS_FOUND)
+    unset(GINKGO_INTERFACE_CFLAGS_FOUND)
+    # Prepare recursively populated library list
+    string(TOUPPER "${CMAKE_BUILD_TYPE}" upper_CMAKE_BUILD_TYPE)
+    if (upper_CMAKE_BUILD_TYPE STREQUAL "DEBUG")
+        list(APPEND GINKGO_INTERFACE_LIBS_FOUND "-lginkgo${CMAKE_DEBUG_POSTFIX}")
+    else()
+        list(APPEND GINKGO_INTERFACE_LIBS_FOUND "-lginkgo")
+    endif()
+    # Prepare recursively populated include directory list
+    list(APPEND GINKGO_INTERFACE_CFLAGS_FOUND
+        "-I${CMAKE_INSTALL_PREFIX}/${GINKGO_INSTALL_INCLUDE_DIR}")
```

```suggestion
        "-I\${includedir}")
```
should it reuse the variable in pc like includedir or something else?